app/diagnostics_test.go

```diff
@@ -273,3 +273,214 @@ func TestDiagnostics(t *testing.T) {
 		}
 	})
 }
+
+func TestRudderDiagnostics(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+
+	th := SetupWithCustomConfig(t, func(config *model.Config) {
+		*config.PluginSettings.Enable = false
+	})
+	defer th.TearDown()
+
+	type payload struct {
+		MessageId string
+		SentAt    time.Time
+		Batch     []struct {
+			MessageId  string
+			UserId     string
+			Event      string
+			Timestamp  time.Time
+			Properties map[string]interface{}
+		}
+		Context struct {
+			Library struct {
+				Name    string
+				Version string
+			}
+		}
+	}
+
+	data := make(chan payload, 100)
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		body, err := ioutil.ReadAll(r.Body)
+		require.NoError(t, err)
+
+		var p payload
+		err = json.Unmarshal(body, &p)
+		require.NoError(t, err)
+
+		data <- p
+	}))
+	defer server.Close()
+
+	diagnosticID := "test-diagnostic-id-12345"
+	th.App.SetDiagnosticId(diagnosticID)
+	th.Server.initRudder(server.URL)
+
+	assertPayload := func(t *testing.T, actual payload, event string, properties map[string]interface{}) {
+		assert.NotEmpty(t, actual.MessageId)
```

Comment:
Let's add a `t.Helper()` here.

Reply:
Is this still worth it once the Segment test is deleted?